extensions-core/druid-catalog/src/main/java/org/apache/druid/catalog/http/CatalogResource.java

```diff
@@ -0,0 +1,747 @@
+   *
+   * @param dbSchema The name of the schema that holds the table.
+   * @param name The name of the table definition to delete. The user must have
+   *             write access.
+   * @param ifExists Optional flag. If {@code false} (the default), 404 (NOT FOUND)
+   *                 error is returned if the table does not exist. If {@code true},
+   *                 then acts like the SQL IF EXISTS clause and does not return an
+   *                 error if the table does not exist,
+   */
+  @DELETE
+  @Path("/tables/{dbSchema}/{name}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response deleteTable(
+      @PathParam("dbSchema") String dbSchema,
+      @PathParam("name") String name,
+      @QueryParam("ifExists") boolean ifExists,
+      @Context final HttpServletRequest req
+  )
+  {
+    TableId tableId = new TableId(dbSchema, name);
+    Pair<Response, SchemaSpec> result = validateSchema(tableId.schema());
+    if (result.lhs != null) {
+      return result.lhs;
+    }
+    SchemaSpec schema = result.rhs;
+    if (!schema.writable()) {
+      return Actions.badRequest(
+          Actions.INVALID,
+          StringUtils.format("Cannot delete tables from schema %s", tableId.schema()));
+    }
+    if (Strings.isNullOrEmpty(name)) {
+      return Actions.badRequest(Actions.INVALID, "Table name is required");
+    }
+    try {
+      catalog.authorizer().authorizeTable(schema, tableId.name(), Action.WRITE, req);
+    }
+    catch (ForbiddenException e) {
+      return Actions.forbidden(e);
+    }
+    try {
+      if (!catalog.tables().delete(tableId) && !ifExists) {
+        return Actions.notFound(tableId.sqlName());
+      }
+    }
+    catch (Exception e) {
+      return Actions.exception(e);
+    }
+    return Actions.ok();
+  }
+
+  public static final String SCHEMA_SYNC = "/schemas/{dbSchema}/sync";
+
+  /**
+   * Synchronization request from the Broker for a database schema. Requests all
+   * table definitions known to the catalog. Used to prime a cache on first access.
+   * After that, the Coordinator will push updates to Brokers. Returns the full
+   * list of table details.
+   *
+   * It is expected that the number of table definitions will be of small or moderate
+   * size, so no provision is made to handle very large lists.
+   */
+  @GET
+  @Path(SCHEMA_SYNC)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response syncSchema(
+      @PathParam("dbSchema") String dbSchema,
+      @Context final HttpServletRequest req
+  )
+  {
+    // Same as the user-command for now. This endpoint reserves the right to change
+    // over time as needed, while the user endpoint cannot easily change.
+    return listTableDetails(dbSchema, req);
+  }
+
+  public static final String TABLE_SYNC = "/tables/{dbSchema}/{name}/sync";
+
+  /**
+   * Synchronization request from the Broker for information about a specific table
+   * (datasource). Done on first access to the table by any query. After that, the
```

Comment:
Why doesnt the coordinator push updates to brokers when table definition is created, why does it need to be triggered by the broker receiving a query for the table? Seems like it could cause slow first queries.